Program.cs

```diff
@@ -1,12 +1,24 @@
 ﻿using MongoDB.Driver;
 using MongoDB.Bson;
 
-const string connectionUri = "mongodb://localhost:27017/";
+public static class Example
+{
+    public static void Main(string[] args)
+    {
+        var settings = MongoClientSettings.FromConnectionString(args[0].ToString());
+        // https://jira.mongodb.org/browse/CSHARP-3516
+        // Set the ServerApi field of the settings object to Stable API version 1
+        // to force OP_MSG to be used in hello command during handshake.
+        settings.ServerApi = new ServerApi(ServerApiVersion.V1);

```

Comment:
I think we should implement `saslStart` OP_QUERY command instead. Setting Stable API version will not allow us to test it.